src/core/core.plugin.js

```diff
@@ -71,23 +71,34 @@ module.exports = function(Chart) {
 		 * returned value can be used, for instance, to interrupt the current action.
 		 * @param {String} extension the name of the plugin method to call (e.g. 'beforeUpdate').
 		 * @param {Array} [args] extra arguments to apply to the extension call.
+		 * @param {ChartInstance} the chart instance for dispatching any inlint plugins.
 		 * @returns {Boolean} false if any of the plugins return false, else returns true.
 		 */
-		notify: function(extension, args) {
-			var plugins = this._plugins;
```

Comment:
I think this would have been easier and shorter:
```javascript
var options = (chart && chart.options) || {};
var plugins = [].concat(this._plugins, options.plugins || []);

// ... previous code ...
```